app/src/main/java/android/example/courses/AttendanceFragment.kt

```diff
@@ -0,0 +1,86 @@
+package android.example.courses
+
+import android.example.courses.dataClasses.Attendance
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import com.firebase.ui.firestore.FirestoreRecyclerAdapter
+import com.firebase.ui.firestore.FirestoreRecyclerOptions
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.CollectionReference
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.android.synthetic.main.fragment_attendance_item.view.*
+
+class AttendanceFragment : Fragment() {
+    private var db = FirebaseFirestore.getInstance()
+    private var auth = FirebaseAuth.getInstance()
+
+    private var attendanceAdapter: FirestoreRecyclerAdapter<Attendance, AttendanceViewHolder>? =
+        null
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        val query = db.collection("attendance/${auth.currentUser?.email}/data")
```

Comment:
@radumarin88: When I tried to read data from firestore and to display it with a recycler view I was a bit confused by this query thing.

I kind of makes sense the way it works: you pass in a query that should return multiple documents and for each document the method  `onBindViewHolder` is called.

But what if you have a document that has an array or a map (as a field) and you want to display that array/map in the recycler view? In that case, you'll basically return one document each time, so you won't be able to build the query (or at least I don't know how to do it).

I'm asking this because initially my data model was the following:

`attendance (collection) --> user@email.com (document) --> attendace: [course1: 4, course2: 7, course3: 10]`

So I just wanted to say `.get(user@email.com)`, to get the `attendance` field and to use it to power up the recyclerView.

I didn't work, so I had to update the data model to:

`attendance (collection) --> user@email.com (document) --> data (collection) --> courseName (document) -> fields `

Basically, I had to add an extra collection (named `data` in that case) and inside it I added a document for each course.

What do you think about that? Is this the right way of doing it?